substrate/primitives/consensus/beefy/src/lib.rs

```diff
@@ -442,6 +442,9 @@ pub trait AncestryHelper<Header: HeaderT> {
 	/// The data needed for validating the proof.
 	type ValidationContext;
 
+	#[deprecated(
+		note = "generate_proof will be removed some time after September 2025. Use mmr::MmrAPI::generate_ancestry_proof instead"
+	)]
```

Comment:
I would remove this method since it's not part of a RPC Api. I would just write in the PRdoc that `mmr::MmrAPI::generate_ancestry_proof` should be used instead. I don't think anyone uses it apart from us anyway.